config/oidc/config.proto

```diff
@@ -75,14 +75,18 @@ message OIDCConfig {
     // Required.
     string callback_uri = 3 [(validate.rules).string.min_len = 1];
 
-    // The JSON JWKS response from the OIDC provider’s `jwks_uri` URI which can be found in
-    // the OIDC provider's
-    // [configuration response](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse).
-    // Note that this JSON value must be escaped when embedded in a json configmap
-    // (see [example](https://github.com/istio-ecosystem/authservice/blob/master/bookinfo-example/config/authservice-configmap-template.yaml)).
-    // Used during token verification.
-    // Required.
-    string jwks = 4 [(validate.rules).string.min_len = 1];
+    oneof jwks_config {
+        // The JSON JWKS response from the OIDC provider’s `jwks_uri` URI which can be found in
+        // the OIDC provider's
+        // [configuration response](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfigurationResponse).
+        // Note that this JSON value must be escaped when embedded in a json configmap
+        // (see [example](https://github.com/istio-ecosystem/authservice/blob/master/bookinfo-example/config/authservice-configmap-template.yaml)).
+        // Used during token verification.
+        // Required.
+        string jwks = 4;
+        
+        string jwks_uri = 17;
```

Comment:
comment doc string for this? semantics as well as how we handle it ("periodic refresh etc")

Reply:
Yes. I will add detail docs in the future.